instructor results view: show a 'view by section' warning for bigger courses #3024

manvijain opened the pull request:
Fixes #3024

LowWeiLin commented on src/main/webapp/jsp/instructorFeedbackResultsTop.jsp:
```diff
@@ -244,6 +244,11 @@
 </form>
+    <%if(data.selectedSection.equals("All") && showAll==false){%>
+        <div id="statusMessage" class="alert alert-warning" style="margin-top:20px;" >
+            This session seem to have a large number of responses. It is recommended to view the results one section at at time.
```
We should not have "magic strings" like this in the jsp. Try creating a constant in Const.java?

LowWeiLin replied:
Also, there is already a standardized status message available.
See the line below: <jsp:include page="<%=Const.ViewURIs.STATUS_MESSAGE%>" />
Try to find a way to use the existing system instead of creating the div and duplicating code.

manvijain commented:
I made the changes to standardize code. Does it look okay now?

LowWeiLin commented:
There is still a repeat of the status div being made. The html to show the status already exists, and it shows status messages from the variable statusToUser. Instead of copying the entire status div, you should add the error into the variable.

LowWeiLin commented:
The statusToUser variable can be found in Action.java. The error message should be added to this variable from InstructorFeedbackResultsPageAction.java, if all sections are shown, and data.bundle.isComplete is fase.

manvijain commented:
There is a block in InstructorFeedbackResultsPageAction.java which has a if block for the two required conditions. I added my statusToUser variable in it and in const.java I changed the type of my message to StatusMessages but on doing that the 
<jsp:include page="<%=Const.ViewURIs.STATUS_MESSAGE%>" /> tag in instructorFeedbackResultsTop.jsp doesn't show the required status message.

I went through the developer manuals UI component to understand the flow of data better. To my understanding the ActionPage(Controller) send data to the PageData(Model) which gives data to jsp page(View). I am confused as to how will the statusToUser bind itself to the View page as it is not sent along with the createAjaxResult function in the block. Please tell me what am I missing on.

manvijain commented:
I went through the developer manual's UI component to understand the flow of data. I understood that the ActionPage (Controller) will send data to DataPage(Model) which will control the jsp Page(View). I am confused as to how will my message bind itself to the jsp page.Please tell me what am I missing.

LowWeiLin commented:
If you take a look at the code in statusMessage.jsp, you can see that the status message being displayed is obtained from a request parameter. This corresponds to the statusToUser variable in Action.java. If you wish to know exactly how they are binded, you will have to look through the code and find where statusToUser is set as a request parameter for the jsp to use.

LowWeiLin commented:
"I added the statusToUser variable in InstructorFeedbackResultsPageAction.java" - the variable is already present in the parent class Action. You do not have to add the variable, just append to it. 
You can take a look at FeedbackQuestionSubmissionEditSaveAction.java to see examples on how a status message is added to the statusToUser variable.

manvijain commented:
Displayed warning using statusToUser. Is it okay now?

LowWeiLin commented on src/main/java/teammates/common/util/Const.java:
```diff
@@ -942,6 +942,7 @@
         public static final String FEEDBACK_RESULTS_NOTHINGNEW = "You have not received any new feedback but you may review your own submissions below.";
         
         public static final String ENROLL_LINE_EMPTY = "Please input at least one student detail.";
+        public static final String FEEDBACK_SESSION_RESULTSSECTIONVIEWWARNING = "This session seems to have a large number of responses. It is recommended to view the results one section at at time.";
```
FEEDBACK_SESSION_RESULTSSECTIONVIEWWARNING can be changed to  FEEDBACK_SESSION_RESULTS_SECTIONVIEWWARNING, to be consistent with similar messages, and moved up one line so the variables are sorted.

LowWeiLin commented:
Yes! Good job. Do take a look at the files changed tab. You will see many lines have changed due to whiatespace/line breaks. We don't want unnecessary changes in the pull request, could you remove them?
Added one comment as well, otherwise it looks good.

manvijain commented:
Thank you! :D I removed the white spaces and added the comment. All good now?

LowWeiLin commented:
There still seems to be unrelated changes in the diff. Take a look using github's diff?

When I said "Added one comment" i meant i added a comment to the code... but its good that you added a comment.


I commented:
FEEDBACK_SESSION_RESULTSSECTIONVIEWWARNING can be changed to FEEDBACK_SESSION_RESULTS_SECTIONVIEWWARNING, to be consistent with similar messages, and moved up one line so the variables are sorted.

manvijain commented:
I made the changes in Const.java and restored the jsp file to original state as it had no changes related to the issue. I let the changes in html files remain for readability. Hope it is fine now.

LowWeiLin commented:
Looks okay now.
